cloud/azure/azure_rm_publicipaddress.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2016 Matt Davis, <mdavis@ansible.com>
+#                    Chris Houseknecht, <house@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: azure_rm_publicipaddress
+
+version_added: "2.1"
+
+short_description: Manage Azure Public IP Addresses.
+
+description:
+    - Create, update and delete a Public IP address. Allows setting and updating the address allocation method and
+      domain name label. Use the azure_rm_networkinterface module to associate a Public IP with a network interface.
+
+options:
+    resource_group:
+        description:
+            - Name of resource group with which the Public IP is associated.
+        required: true
+    allocation_method:
+        description:
+            - Control whether the assigned Public IP remains permanently assigned to the object. If not
+              set to 'Static', the IP address my changed anytime an associated virtual machine is power cycled.
+        choices:
+            - Dynamic
+            - Static
+        default: Dynamic
+        required: false
+    domain_name_label:
+        description:
+            - The customizable portion of the FQDN assigned to public IP address. This is an explicit setting. If
+              no value is provided, any existing value will be removed on an existing public IP.
+        aliases:
+            - domain_name_label
+        required: false
+        default: null
+    name:
+        description:
+            - Name of the Public IP.
+        required: true
+    state:
+        description:
+            - Assert the state of the Public IP. Use 'present' to create or update a and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+        required: false
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+        default: resource_group location
+        required: false
+    tags:
+        description:
+            - "Dictionary of string:string pairs to assign as metadata to the object. Metadata tags on the object
+              will be updated with any provided values. To remove tags use the purge_tags option."
+        required: false
+        default: null
+    purge_tags:
```

Comment:
I better option would be one that toggles between appending tags or making it a canonical list. I know this follows the aws modules, but that was a bad name there too.